### PR TITLE
Modifications to allow for publish to NPM

### PR DESCRIPTION
**Pull request by hatched**
To QA

`npm install @canonical/jujulib@beta` and then inspect the source, it should only have `js` and `d.ts` files as well as the documentation, examples, and source.

### CODE REVIEW

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: with a suggestion in the code

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ node_modules/
 generator/*.js
 templates/*.js
 api/**/*.js
+api/dist/
```
Comment (anthonydillon) [inline .gitignore:7]: The command below cd into API then outputs back a directory and so isn't dist root level?

**Review by anthonydillon — COMMENTED** [review]
You shouldn't need to publish the .gitignore file to npm

File: .gitignore (hunk 1 above)
Comment (hatched) [reply .gitignore:7]: Good catch, this was a leftover from a previous implementation.

**Comment by hatched** [comment]
Thanks @anthonydillon comments have been addressed 👍

**Pull request merged**